development/extensions/tutorial_testing.rst

```diff
@@ -0,0 +1,913 @@
+
+    install:
+      - composer install --dev --no-interaction --prefer-source
+      - travis/prepare-phpbb.sh $EXTNAME $PHPBB_BRANCH
+      - cd ../../phpBB3
+      - travis/prepare-extension.sh $EXTNAME $PHPBB_BRANCH
+      - travis/setup-phpbb.sh $DB $TRAVIS_PHP_VERSION
+
+    before_script:
+      - travis/setup-database.sh $DB $TRAVIS_PHP_VERSION
+
+    script:
+      - sh -c "if [ '$SNIFF' != '0' ]; then travis/ext-sniff.sh $DB $TRAVIS_PHP_VERSION $EXTNAME; fi"
+      - sh -c "if [ '$IMAGE_ICC' != '0' ]; then travis/check-image-icc-profiles.sh $DB $TRAVIS_PHP_VERSION; fi"
+      - phpBB/vendor/bin/phpunit --configuration phpBB/ext/$EXTNAME/travis/phpunit-$DB-travis.xml --bootstrap ./tests/bootstrap.php
+      - sh -c "if [ '$EPV' != '0' ] && [ '$TRAVIS_PHP_VERSION' = '5.3.3' ] && [ '$DB' = 'mysqli' ]; then phpBB/ext/$EXTNAME/vendor/bin/EPV.php run --dir='phpBB/ext/$EXTNAME/'; fi"
+
+.. note::
+
+    You should not need to make any changes in this file, apart from the
+    following section, which should be quite self explaining:
+
+    .. code-block:: yaml
+
+        env:
+          global:
+            - EXTNAME="acme/demo"  # CHANGE name of the extension HERE
+            - SNIFF="1"            # Should we run code sniffer on your code?
+            - IMAGE_ICC="1"        # Should we run icc profile sniffer on your images?
+            - EPV="1"              # Should we run EPV (Extension Pre Validator) on your code?
+            - PHPBB_BRANCH="3.1.x"
+
+Preparing phpBB
+---------------
+
+The second file we need to create is a helper file called
+``travis/prepare-phpbb.sh``, that is used by Travis CI, to set up the phpBB
+installation from GitHub for us:
+
+.. warning::
+
+    You should not edit the content of this file!
+
+.. code-block:: bash
+
+    #!/bin/bash
+    #
+    # This file is part of the phpBB Forum Software package.
+    #
+    # @copyright (c) phpBB Limited <https://www.phpbb.com>
+    # @license GNU General Public License, version 2 (GPL-2.0)
+    #
+    # For full copyright and license information, please see
+    # the docs/CREDITS.txt file.
+    #
+    set -e
+    set -x
+
+    EXTNAME=$1
+    BRANCH=$2
+    EXTPATH_TEMP=$3
+
+    # Copy extension to a temp folder
+    mkdir ../../tmp
+    cp -R . ../../tmp
+    cd ../../
+
+    # Clone phpBB
+    git clone --depth=1 "git://github.com/phpbb/phpbb.git" "phpBB3" --branch=$BRANCH
+
+Enable Travis CI on GitHub
+--------------------------
+
+As a last step you need to enable Travis CI on GitHub.
+
+    1. Open your repository, e.g. `<https://github.com/phpbb/phpbb-ext-acme-demo>`_,
+    2. Go to "Settings"
+    3. "Webhooks & Services"
+    4. In the "Services" table press the "Add Service" button and search for ``Travis CI``
+
+When you now commit and push your the travis files from above to your ``master``
+branch, the unit, database and functional tests will be executed.
+This helps to avoids have a regression, so basically breaking another part of
```

Comment:
Tests help to avoid causing a regression (breaking another part of your code) while fixing bugs or adding new features and other changes to your extension.